canonical-data-model/canonical-data-model.bal

```diff
@@ -0,0 +1,142 @@
+import ballerina/http;
+import ballerina/io;
+
+const PROJECT_ID = "LAND-TEST04";
+const DATASET_ID = "f57074a0-a8b6-403e-9df1-e9fc46";
+
+final http:Client gMapClient = check new ("http://mapsplatformdatasets.googleapis.com.balmock.io");
+final http:Client microsoftClient = check new ("http://atlas.microsoft.com.balmock.io");
+
+type Gpx xml;
+type Kml xml;
+type Csv string[][];
+
+type GeoJson record {
+    string 'type = "FeatureCollection";
+    string name = "PlaceMarks";
+    Feature[] features;
+};
+
+type Feature record {|
+    string 'type = "Feature";
+    record {|
+        string X;
+        string Y;
+        string Name;
+        string description;
+    |} properties;
+    record {|
+        string 'type = "Point";
+        string[] coordinates;
+    |} geometry;
+|};
+
+type GMapResponse record {
+    string name;
+    string displayName;
+    string createTime;
+};
+
+type MicrosoftMapResponse record {
+    string formatVersion;
+    record {
+        record {
+            string latitude;
+            string longitude;
+        }[] points; 
+    }[] routes;
+};
+
+type A record {};
+
+service /map on new http:Listener(8080) {
+    resource function post uploadPlaceMarks(http:Request request) returns GMapResponse|error? {
+        Csv|Gpx data = check getPayload(request);
+        GeoJson geoJson = convertToCanonical(data);
+        Kml kmlData = convertFromCanonicalToKml(geoJson);
+        GMapResponse gMapResponse = check gMapClient->post(
+            string `/v1/projects/${PROJECT_ID}/datasets/${DATASET_ID}:import`, kmlData);
+        return gMapResponse;
+    }
+
+    resource function post getRouteDirection(http:Request request, string query) returns MicrosoftMapResponse|error {
+        Csv|Gpx data = check getPayload(request);
+        GeoJson geoJson = convertToCanonical(data);
+        MicrosoftMapResponse microsoftMapResponse = check microsoftClient->/route/directions/'json.post(
+            {"supportingPoints": geoJson.toJson()}, {query : string `${query}`});
+        return microsoftMapResponse;
+    }
+}
+
+isolated function convertToCanonical(Csv|Gpx data) returns GeoJson {
+    if data is Csv {
+        return convertFromCsvToCanonical(data);
+    } else {
+        return convertFromGpxToCanonical(data);
+    }
+}
+
+isolated function getPayload(http:Request request) returns Csv|Gpx|error {
```

Comment:
Added this to handle csv payload since it will be received as serialized byte[].